python/core/auto_generated/geometry/qgsgeometry.sip.in

```diff
@@ -1704,7 +1704,7 @@ Returns a null geometry on exception.
 .. versionadded:: 3.20
 %End
 
-    QgsGeometry subdivide( int maxNodes = 256 ) const;
+    QgsGeometry subdivide( int maxNodes = 256, double gridSize = -1 ) const;
```

Comment:
Thought: Rather than adding a specific "gridSize" parameter to all these methods, I'd lean toward making a small "QgsGeometryOptions" class/struct which contains a gridSize member and pass that to these methods instead. We could then easily extend that class/struct with additional options in future (eg default m/z values)